test/blackbox-tests/test-cases/github2681/run.t

```diff
@@ -0,0 +1,31 @@
+A rule may have an alias field. This denotes that the action of the rule is a
+dependency of the alias.
+  $ mkdir simple && cd simple
+  $ cat > dune-project <<EOF
+  > (lang dune 2.0)
+  > EOF
+  $ cat > dune <<EOF
+  > (rule
+  >  (action (with-stdout-to foo (echo "hello world")))
+  >  (alias bar))
+  > EOF
+  $ dune build @bar --display short
+  $ cat _build/default/foo
+  hello world
+
+The field order should not affect the alias field:
```

Comment:
Is this really worth testing? Isn't the field parser already making sure of that?

Reply:
Could you add a test with non-empty set of targets as well (maybe instead of the reordering test)?

Reply:
I removed the extra test. In fact, my parsing code was a little wrong before so this failed.

I assume you meant an empty set of targets. I added such a test.